x/interchainqueries/keeper/msg_server.go

```diff
@@ -77,6 +81,74 @@ func (k msgServer) RegisterInterchainQuery(goCtx context.Context, msg *types.Msg
 	return &types.MsgRegisterInterchainQueryResponse{Id: lastID}, nil
 }
 
+func (k msgServer) RemoveInterchainQuery(goCtx context.Context, msg *types.MsgRemoveInterchainQueryRequest) (*types.MsgRemoveInterchainQueryResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	ctx.Logger().Debug("RemoveInterchainQuery", "msg", msg)
+
+	if err := msg.ValidateBasic(); err != nil {
+		ctx.Logger().Debug("RemoveInterchainQuery: failed to validate message", "message", msg)
+		return nil, sdkerrors.Wrapf(err, "invalid msg: %v", err)
+	}
+
+	query, err := k.GetQueryByID(ctx, msg.GetQueryId())
+	if err != nil {
+		ctx.Logger().Debug("RemoveInterchainQuery: failed to GetQueryByID",
+			"error", err, "query_id", msg.QueryId)
+		return nil, sdkerrors.Wrapf(err, "failed to get query by query id: %v", err)
+	}
+	if query.GetOwner() != msg.GetSender() {
+		ctx.Logger().Error("RemoveInterchainQuery: authorization failed",
+			"msg", msg)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "authorization failed")
+	}
+	k.RemoveQueryByID(ctx, query.Id)
+	if query.GetQueryType() != TransactionsQueryType {
```

Comment:
We don't have query type `x/tx/RecipientTransactions` anymore. 
I guess what you want to do is:
```go
if types.InterchainQueryType(query.GetQueryType()).IsKV() {
```